apps/server/src/connect/userEndpoints.ts

```diff
@@ -0,0 +1,35 @@
+import type { Response } from 'express'
+import { getProviderAdapter } from '../adapters'
+
+interface UserDeleteQueryParameters {
+  provider: string
+}
+
+export interface UserDeleteRequest {
+  query: UserDeleteQueryParameters
```

Comment:
It feels weird to have one of these parameters come from the query and one come from the route params

Reply:
This is true thinking about this some more and doing some research I think it might have been a mistake to use query params so heavily with the data endpoints. It seems like most commonly with rest apis, route params are used for params which are required and query params are used for optional params like pagination type things or sort order. 

So the delete endpoint might look like:
```javascript
app.delete(`${ApiEndpoints.USER}/:provider/:userId`, userDeleteHandler)
```

And the data endpoints might change to look like:
```javascript
app.get('/:provider/data/:userId/:connectionId/accounts', accountsDataHandler)
```
For the transactions endpoint I think `start_time` and `end_time` would be query params

Reply:
I think it'd probably be good to refactor these endpoints a bit to follow the common rest patterns. I'll make a ticket to do that for this endpoint and the data endpoints.